sqltap/sqltap.py

```diff
@@ -296,6 +338,8 @@ def __init__(self, stats, report_file=None, report_dir=".",
 
         :param template_dir: folder of the template to generate the report.
         """
+        self.duration = ((stats[-1].end_time - stats[0].start_time)
+                         if stats else 0)
```

Comment:
it's probably worth the profiling session recording its own start time and end time since this calculation is not really accurate. if there is a 15-minute profiling session with only a single query captured, this duration will report 0. (worse, if there are no queries captured at all, it will throw an exception)